### PR TITLE
feat: Modified Equinoctial Orbital Elements (MEOE)

### PR DESCRIPTION
**Pull request by vishwa2710**
Generated by Jules google agent
Testing out.
---
This commit introduces the Modified Equinoctial Orbital Elements (MEOE) class, providing an alternative to Classical Orbital Elements (COE) for representing satellite orbits, particularly useful for orbits with small eccentricities or inclinations.

The MEOE set includes:
- p: semi-latus rectum
- f: x-component of eccentricity vector (e * cos(RAAN + AOP))
- g: y-component of eccentricity vector (e * sin(RAAN + AOP))
- h: x-component of node vector (tan(i/2) * cos(RAAN))
- k: y-component of node vector (tan(i/2) * sin(RAAN))
- L: true longitude (RAAN + AOP + True Anomaly)

Key features implemented:
1.  **MEOE Class:**
    *   Defined in `OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Model/MEOE.hpp` and implemented in `MEOE.cpp`.
    *   Includes constructors, getters for each element, equality operators, and print/string representations.
2.  **Cartesian Conversions:**
    *   `MEOE::Cartesian()`: Converts Cartesian state (position, velocity) to MEOE. This is achieved by first converting to COE and then to MEOE, handling singularities like 180-degree inclination.
    *   `MEOE::getCartesianState()`: Converts MEOE to Cartesian state using direct formulas from JPL documentation.
3.  **Unit Tests (C++):**
    *   Added `MEOE.test.cpp` with comprehensive tests for constructors, getters, conversions (including round-trip), and edge cases (circular, equatorial, 180-deg inclination orbits, p<=0, q=0 singularities).
4.  **CMake Integration:**
    *   The new files are automatically included in the build process due to existing `FILE(GLOB_RECURSE)` patterns in CMakeLists.txt.
5.  **Python Bindings:**
    *   Exposed the `MEOE` class and its methods to Python via Pybind11.
    *   Created `bindings/python/src/OpenSpaceToolkitAstrodynamicsPy/Trajectory/Orbit/Model/MEOE.cpp` for the binding code.
    *   Updated Python binding CMakeLists to correctly include `.cpp` source files.
6.  **Python Tests:**
    *   Added `bindings/python/test/trajectory/orbit/models/test_meoe.py` with a comprehensive suite of tests mirroring the C++ tests, ensuring the Python interface is robust and correct.

The implementation aims to be consistent with the existing COE class structure and provides a valuable new tool for orbital mechanics analysis within OpenSpace Toolkit.

## Summary by CodeRabbit

- **New Features**
  - Introduced support for Modified Equinoctial Orbital Elements, allowing users to create, access, and convert orbits using this representation.
  - Added Python bindings for the Modified Equinoctial orbit model, enabling use and manipulation directly from Python.
- **Tests**
  - Added comprehensive unit tests for Modified Equinoctial orbits in both C++ and Python to ensure correctness and reliability.

### CODE REVIEW

File: include/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Model/MEOE.hpp [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+using ostk::physics::unit::Length;
+using ostk::physics::unit::Angle;
+using ostk::physics::unit::Derived;
+using ostk::physics::time::Instant;
+using ostk::physics::environment::object::Celestial;
+using ostk::physics::coordinate::Frame;
+
+/// @brief Modified Equinoctial Orbital Elements (MEOE)
+///
+/// @ref https://spsu.edu/math/npetulante/mathematica/AppliedOrbitalMechanicsSupplement.pdf
```
Comment (alex-liang3) [inline include/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Model/MEOE.hpp:48]: this is a dead link for me

File: include/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Model/MEOE.hpp [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    enum class Element
+    {
+        SemiLatusRectum,  ///< Semi-latus rectum (p)
+        F,                ///< x-component of eccentricity vector (f)
+        G,                ///< y-component of eccentricity vector (g)
+        H,                ///< x-component of node vector (h)
+        K,                ///< y-component of node vector (k)
+        TrueLongitude     ///< True longitude (L)
+    };
+
```
[the remark this answers is not in the document]
Comment (alex-liang3) [reply include/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Model/MEOE.hpp:165]: the bot's got a point - this should probably go at the top of the class?

File: src/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Model/MEOE.cpp [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+
+bool MEOE::operator==(const MEOE& aMEOE) const
+{
+    if ((!this->isDefined()) || (!aMEOE.isDefined()))
+    {
+        return false;
+    }
+
+    // Comparison with tolerance for Real and Angle types
+    const Real tolerance = Real::Epsilon() * 100.0; // Define an appropriate tolerance
```
Comment (alex-liang3) [inline src/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Model/MEOE.cpp:48]: should this match the tolerances in the python tests?

File: include/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Model/Kepler/ModifiedEquinoctial.hpp [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+using ostk::physics::time::Instant;
+using ostk::physics::unit::Angle;
+using ostk::physics::unit::Derived;
+using ostk::physics::unit::Length;
+
+using KeplerianCOE = ostk::astrodynamics::trajectory::orbit::model::kepler::COE;
+
+/// @brief Modified Equinoctial Orbital Elements (ModifiedEquinoctial)
+///
+/// @ref https://spsu.edu/math/npetulante/mathematica/AppliedOrbitalMechanicsSupplement.pdf
```
[the remark this answers is not in the document]
Comment (vishwa2710) [reply include/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Model/Kepler/ModifiedEquinoctial.hpp:54]: Weird I had updated it, maybe I missed one. Will fix!

File: src/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Model/Kepler/ModifiedEquinoctial.cpp [hunk 5]
```diff
@@ -0,0 +1,372 @@
...
+    const Real vx_pf = -muOverPSqrt * (sinL + g_val);
+    const Real vy_pf = muOverPSqrt * (cosL + f_val);
+
+    // Inertial frame components
+
+    const Real x = sSquaredInverse * (x_pf * (1.0 + alpha_squared) + 2.0 * h_val * k_val * y_pf);
+    const Real y = sSquaredInverse * (x_pf * (2.0 * h_val * k_val) + y_pf * (1.0 - alpha_squared));
+    const Real z = sSquaredInverse * (-2.0 * k_val * x_pf + 2.0 * h_val * y_pf);
+
+    const Position position = Position::Meters({x, y, z}, aFrameSPtr);
```
[the remark this answers is not in the document]
Comment (vishwa2710) [reply src/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Model/Kepler/ModifiedEquinoctial.cpp:214]: Yes indeed. I added a check and some tests.

**Review by phc1990 — APPROVED** [review]
Left 2 nits

File: src/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Model/Kepler/ModifiedEquinoctial.cpp [hunk 6]
```diff
@@ -0,0 +1,423 @@
...
+        angularMomentumMagnitude;
+
+    // Modified equinoctial elements nodeY and nodeX
+    const Real nodeX = angularMomentumDirection.x() / (1.0 + angularMomentumDirection.z());
+    const Real nodeY = -angularMomentumDirection.y() / (1.0 + angularMomentumDirection.z());
+
+    // Check for singularity (angularMomentumDirection.z() == -1, i.e., inclination = 180 degrees)
+    if (std::abs(1.0 + angularMomentumDirection.z()) < Real::Epsilon())
+    {
+        return ModifiedEquinoctial::Undefined();
```
[the remark this answers is not in the document]
Comment (vishwa2710) [reply src/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Model/Kepler/ModifiedEquinoctial.cpp:309]: It allows the user to decide what to do with the output, raise their own error or manage it in a different way.

[the remark this answers is not in the document]
Comment (apaletta3) [reply src/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Model/Kepler/ModifiedEquinoctial.cpp:309]: It fits into our pattern of undefined objects in OSTk so fair enough - but I would like to start moving away from those undefined objects, because they are kind of useless. What can a user actually do with an Undefined objects other than raise an error cause an undefined object won't work with any of their logic
